### ISSUE TITLE
Some game is too slow.....

### ISSUE THREAD

**Issue by bigly2001** [libretro/libretro-uae#290]
Hi
Why Some games like shadow of the beast III or Second samurai is too slow on my raspberry pi 3 board Retropie 4.6 while another game like Turrican 2 or 3 works great.

**Comment by sonninnos** [comment]
Not all games utilize the hardware the same amount. Try underclocking the CPU option for example and make sure "Cycle-exact" is not used. Other than that I'm afraid the core isn't going to get any faster.

**Comment by bigly2001** [comment]
ok thx
I am sure cycle exact is not used.
 but on other amiga emulator like amiberry shadow of the beast III and ... work great as turrican 2.

**Comment by sonninnos** [comment]
Any idea if the UAE version in Amiberry is older or newer than this 2.6.1? Older would explain the fastness by simply being less accurate. Or if it really is some kind of optimization, it sure would be nice to bring here also. But that would have to be investigated and operated by someone who has the necessary hardware.

**Comment by bigly2001** [comment]
thx for your replay
I have a question because idont know it.
I want to know if puae use the latest core of winuae or if it uses its owns core ?

**Comment by WindyWinston** [comment]
@sonninnos I think Amiberry was originally based off **UAE4Arm** (which I think started on the 'Pandora' handheld). This guy was the main contributor of that https://github.com/PandTomB/uae4arm
Amiberry has used most of his code and other fixes from WinUAE. I think it's mostly just hacks to get things to work and i'm not even sure it has Cycle Accuracy to this day. Maybe there is something worth borrowing hackwise.

**Comment by sonninnos** [comment]
> I want to know if puae use the latest core of winuae or if it uses its owns core ?

No. This PUAE is like I said based on UAE version 2.6.1, and WinUAE currently is 4.3. Although there are many parts I've updated from the current WinUAE, like sound parts are almost completely modern, and many other things needed for the CD32 that was missing here. And some small things even WinUAE does not have.

So it is infact not 2.6.1 anymore, and there isn't much benefit calling it any other, because the only real option going properly forward would be making a new core based on either current FS-UAE or WinUAE. All options have their obstacles that I'm not going to explain now. And having the latest and the most accurate also very likely means having the slowest..



> I think Amiberry was originally based off UAE4Arm

Would indeed be nice if some ARM enthusiast would have a look at both of these and see if there is something worthy. I only have a RPI2 to test with, and I don't exactly have unlimited time on my hands even if it may sometimes seem that way, heh.

**Comment by i30817** [comment]
Sometimes it's a good idea to force a A1200 (sometimes it breaks).

**Comment by Chips-fr** [comment]
There is a uae4arm libretro core that has been updated lately:
https://github.com/libretro/uae4arm-libretro
It's based on old uae4arm version and mounting lha files containing whdload is not supported but hdf & adf is ok...
Integrate ARM Jit and arm optimizations (asm code) from uae4arm into this repo should be doable but needs good insight how jit works...

**Comment by sonninnos** [comment]
I've tried multiple times to get JIT support to even compile here, but it is promising if it works there.

**Comment by WindyWinston** [comment]
This would be great if JIT support could be ported. It makes WHDLoad Demos run fast and smooth even on slow devices. Not much use for games.

**Comment by bigly2001** [comment]
Hi 
I purchased asus tinker board and since the performance of this board is twice as raspberry pie 3 board the problem solved.

**Comment by Frakatchoo** [comment]
Now that I can launch the games on my android devices, I must say that I am quite disappointed with the performance of the core. The games are mostly very slow, with a lot of lag and audio stutters. Even on a snapdragon 845, it doesn't run well.

The comparison with my RPI3 under amiberry is quite obvious.

I hope that optimizations will happen ;)

**Comment by WindyWinston** [comment]
It's not as fast as Amiberry because Amiberry uses a lot of hacks. PUAE is a lot more accurate. Are you sure you don't have the Cycle Exact option on ? My GPDXD plays nearly all the ECS games I like to play at full speed. Only some AGA games and ones that use copper effects have slight slowdown. JIT won't speed these up. JIT is only good for games and demos that use 3D, like Gloom etc.

**Comment by rsn8887** [comment]
I confirm slow performance on RK3326 devices as well (I have Powkiddy RGB10Max2, but all those devices are similar). For example, just displaying the Workbench v1.3 kickstart screen gives only 45 fps. So no game runs at full speed. Cycle exact is already disabled, threaded video and all other speed-enhancing options are already used. Something is going on, because the performance was much faster before the update. I suspect some "hidden" (not exposed to the Core Options panel) is causing slowness? I am not sure, but why would the performance go down so much?

**Comment by sonninnos** [comment]
Yes, the current 4.5.0 is much slower.

**Comment by NgrNxk** [comment]
Are you playing the WHDLoad versions by any chance? At least for me on my RGB20SX setting the machine to A600 helped here. Normally for WHDLoad files, PUAE 2021 uses the A1200-setup. It seems that this doesn't always work very well. As I didn't want to change the automatic selection nor the default for "Automatic HD setup", I just add a " (A600)" to the lha-files that are running slow and the issue is gone.
